state/core_access.go

```diff
@@ -55,7 +55,6 @@ type CoreAccessor struct {
 	keyring keyring.Keyring
 	client  *user.TxClient
 
-	// TODO: remove in scope of https://github.com/celestiaorg/celestia-node/issues/3515
```

Comment:
IMO better to remove when 3515 will be closed.

Reply:
3515 is not relevant anymore. I will close it once this pr is merged with the explanation.
TLDR: you can't rely on the txClient if it is not initialized(so you can't get the default address and account).